src/Dhall/TH.hs

```diff
@@ -0,0 +1,28 @@
+{-# LANGUAGE TemplateHaskell #-}
+
```

Comment:
Could you add a module header explaining an example use case for this functionality?